gandalf.js

```diff
@@ -1,119 +1,120 @@
 /**
```

Comment:
SCRIPT-Well done with the efficient error handling, it covers all the possible scenarios.